File: LeetCode/1207.cpp
//Link: https://leetcode.com/problems/unique-number-of-occurrences/

class Solution {
private:
public:
    bool uniqueOccurrences(vector<int>& arr) {
        set<int> unique(arr.begin(), arr.end());
        set<int> unique_count;
        for(int num : unique)
            unique_count.insert(count(arr.begin(), arr.end(), num));
        return unique.size() == unique_count.size();
    }
};

File: LeetCode/py/1207.py
#Link: https://leetcode.com/problems/unique-number-of-occurrences/

class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        unique = set(arr)
        unique_count = set()
        for num in unique:
            unique_count.add(arr.count(num))
        return len(unique) == len(unique_count)
